app/lib/synthetic_study_populator.rb

```diff
@@ -185,52 +184,51 @@ def self.process_mtx_file_params(study, file_info)
     params = {}
     if file_info['type'] == '10X Barcodes File' || file_info['type'] == '10X Genes File'
       matrix_name = file_info['matrix_file_name']
-      if !matrix_name
+      unless matrix_name
         throw 'Barcodes and Genes files must specify a matrix_file_name'
       end
-      matching_mtx_file = StudyFile.find_by(name: matrix_name, study: study)
+      matching_mtx_file = StudyFile.find_by(name: matrix_name, study:)
       if matching_mtx_file.nil?
         throw "No MTX file with name #{matrix_name} to match #{file_info['filename']}"
       end
-      params[:options] = {'matrix_id' => matching_mtx_file.id.to_s}
+      params[:options] = { 'matrix_id' => matching_mtx_file.id.to_s }
     end
     params
   end
 
   def self.process_sequence_file_params(study, file_info)
     params = {}
     if file_info['type'] == 'BAM Index'
-      if !file_info['bam_file_name']
+      unless file_info['bam_file_name']
         throw 'BAM index files must specify a bam_file_name'
       end
-      matching_bam_file = StudyFile.find_by(name: file_info['bam_file_name'], study: study)
+      matching_bam_file = StudyFile.find_by(name: file_info['bam_file_name'], study:)
       if matching_bam_file.nil?
         throw "No BAM file with name #{file_info['bam_file_name']} to match bai file"
       end
-      params[:options] = {'bam_id' => matching_bam_file.id.to_s}
+      params[:options] = { 'bam_id' => matching_bam_file.id.to_s }
     end
     params
   end
 
-
   # process coordinate/cluster arguments, return a hash of params suitable for passing to a StudyFile constructor
   def self.process_coordinate_file_params(study, file_info)
     params = {}
-    if !file_info['is_spatial'].nil?
+    unless file_info['is_spatial'].nil?
       params[:is_spatial] = file_info['is_spatial']
     end
     if file_info['spatial_cluster_associations'].present?
       # look up the ids for the associations
       # note that this requires the associated file to have already been added
       params[:spatial_cluster_associations] = file_info['spatial_cluster_associations'].map do |cluster_file_name|
-        StudyFile.find_by!(study: study, name: cluster_file_name).id.to_s
+        StudyFile.find_by!(study:, name: cluster_file_name).id.to_s
       end
     end
     params
   end
 
   # process species/annotation arguments, return a hash of params suitable for passing to a StudyFile constructor
-  def self.process_genomic_file_params(study, file_info)
+  def self.process_genomic_file_params(_study, file_info)
```

Comment:
And similarly in callers, since this parameter is unused.

```suggestion
  def self.process_genomic_file_params(file_info)
```

Reply:
5d5213403